pennylane/ops/op_math/composite.py

```diff
@@ -156,9 +156,11 @@ def eigendecomposition(self):
                 eigenvectors of the operator.
         """
         if self.hash not in self._eigs:
-            Hmat = self.matrix()
-            Hmat = math.to_numpy(Hmat)
-            w, U = np.linalg.eigh(Hmat)
+            mat = (
+                self.matrix()
+            )  # we no longer check for hermiticity and allow "anything" on simulator
+            mat = math.to_numpy(mat)
+            w, U = np.linalg.eig(mat)
```

Comment:
We could do an `if self.is_hermitian` check and use the more efficient `eigh` in that case.

Reply:
Added 👍🏼